x/nft/types/nft.go

```diff
@@ -9,11 +9,17 @@ import (
 	sdk "github.com/cosmos/cosmos-sdk/types"
 )
 
-// TODO: create interface for buyable NFT
-
 // NFT non fungible token interface
 type NFT interface {
-	EditMetadata(editName, editDescription, editImage, editTokenURI bool, name, description, image, tokenURI string)
+	GetID() uint64
+	GetOwner() sdk.AccAddress
+	GetName() string
+	GetDescription() string
+	GetImage() string
+	GetTokenURI() string
+
+	EditMetadata(name, description, image, tokenURI string)
+	String() string
```

Comment:
@fedekunze having an interface like this would allow apps to change where/how the id, owner, name etc are returned, right? this would allow custom kvstore behavior right?

Reply:
so the interface just defines the basic set of functions (args and return values) that need to be part of an NFT implementation. The user can extend these functionalities by creating extra functions for this basic NFT implementation